src/modules/jcms_notifications/src/FetchArticleService.php

```diff
@@ -63,6 +63,43 @@ public function getArticle(string $id) {
   }
 
   /**
+   * Gets the IDs for every article in Lax.
+   *
+   * @return array
+   */
+  public function getAllArticleIDs() {
+    $ids = [];
+    $endpoint = Settings::get('jcms_all_articles_endpoint');
+    if ($endpoint) {
+      $auth_key = 'jcms_article_auth_unpublished';
+      $stop = FALSE;
+      while (!$stop) {
+        $page = $page ?? 1;
+        $response = $this->client->get($endpoint, ['auth' => Settings::get($auth_key), 'http_errors' => FALSE, ['query' => ['page' => $page]]]);
+        if ($response instanceof ResponseInterface) {
+          $body = $response->getBody()->getContents();
+          $json = json_decode($body, TRUE);
+          if (isset($json['items']) && !empty($json['items'])) {
+            foreach ($json['items'] as $data) {
+              if (isset($data['id'])) {
+                $ids[] = $data['id'];
+                $page++;
```

Comment:
Not quite sure I'm getting what this page variable is? Is it incrementing every single item, or just every set of items? It looks like it might be needed outside of the foreach